#2442 fixed data explorer time range buttons

IsaakKrut opened the pull request:
Fixes #2442 

PR introduces (a) breaking change(s): no

PR introduces (a) deprecation(s): no

IsaakKrut commented:
Is there a reason why 1 month offset was 60*24*30 + 600 minutes? Updated it to be exactly 30 days

tenthe commented on ui/cypress/tests/datalake/timeRangeSelectors.spec.ts:
```diff
@@ -0,0 +1,137 @@
+import { DataLakeUtils } from '../../support/utils/datalake/DataLakeUtils';
+
+describe('Test Time Range Selectors in Data Explorer', () => {
+    const periods = [
```
You could use a list of objects here instead of a list of stings, then you can remove the switch block in `getExpectedStartDate`

tenthe reviewed (commented):
Hi @IsaakKrut,
Thank you for your PR. I have checked and it worked as expected. 
As for your question about the one month delay, I'm not sure why that was the case. Your change makes perfect sense to me.

I have some minor comments for possible code improvements what do you think about them?

I really like the fact that you have directly provided an e2e test for this function. For me the test was a bit hard to read, is it possible to refactor it a bit to make the individual steps clearer.
One reason I found it difficult may be that I am not familiar with the Cypress `invoke` keyword. How does it work exactly?

Thanks again for your contribution.

IsaakKrut commented:
Hi @tenthe,

Updated the test with your suggestions and fixed a daylight savings related issue. 

Regarding invoke method. What it does is it invokes a function specified in the first parameter on an object yielded before it in the chain. In this case, it's an attr() function that gets the attribute's value based on the second parameter which is attributeName

This is the html element:
`<sp-date-input data-cy="time-range-to" _nghost-ng-c1192185161="" ng-reflect-date="Mon Mar 11 2024 16:23:16 GMT-0"><input _ngcontent-ng-c1192185161="" type="datetime-local" step="1" class="sp-date-input"></sp-date-input>`

So the current date selection is stored in the "ng-reflect-date" attribute. I added a couple of comments, let me know if I can do anything else to make it easier to read
